boa/src/value/mod.rs

```diff
@@ -527,11 +527,25 @@ impl Value {
         // 1. Assert: input is an ECMAScript language value. (always a value not need to check)
         // 2. If Type(input) is Object, then
         if let Value::Object(obj) = self {
+            if let Some(exotic_to_prim) =
+                obj.get_method(context, context.well_known_symbols().to_primitive_symbol())?
+            {
+                let hint = match preferred_type {
+                    PreferredType::String => "string",
+                    PreferredType::Number => "number",
+                    PreferredType::Default => "default",
+                }
+                .into();
+                let result = context.call(&exotic_to_prim.into(), &self, &[hint])?;
```

Comment:
```suggestion
                let result = exotic_to_prim.call(&self, &[hint], context)?;
```
It's better to use the `GcObject::call()` method instead of `Context::call()` so we don't convert the already `GcObject` to `Value`. Besides this, the PR looks good to me! :)

Reply:
Thanks for the suggestion. Done!